server/infinispan/src/main/java/org/hawkular/apm/server/infinispan/InfinispanCommunicationDetailsCache.java

```diff
@@ -132,4 +135,14 @@ public void store(String tenantId, List<CommunicationDetails> details) throws Ca
         }
     }
 
+    @Override
+    public List<CommunicationDetails> getById(String tenantId, String id) {
+        QueryFactory<?> queryFactory = Search.getQueryFactory(communicationDetails);
+        Query query = queryFactory.from(CommunicationDetails.class)
+                .having("id")
+                .eq(id)
```

Comment:
`tenantId` is not used in any of the cache implementations. For `id` I have added null check.